site/content/entry/yuzu-progress-report-nov-2023/index.md

```diff
@@ -0,0 +1,359 @@
+If you hate clicking your own game out of focus, you will like this change.
+
+While continuing his work on improving yuzu’s file system emulation, byte[] solved a bug that caused `MONSTER HUNTER GENERATIONS ULTIMATE` to {{< gh-hovercard "11936" "fail to boot" >}} when mods for the game were installed.
+
+But that was only an appetiser; here’s the main dish.
+byte[] {{< gh-hovercard "12208" "improved the performance" >}} of parsing and building RomFS mods, lowering the patch time of `The Legend of Zelda: Tears of the Kingdom` with mods applied from 2.8 seconds to 0.3 seconds on a high performance CPU (a Ryzen 9 5950X in this case). This was very noticeable in testing.
+
+Modders out there, fill up that list–you won’t waste time now waiting for games to load.
+
+Exactly the same fix needed for `Super Mario RPG` but in a different spot allows `Star Ocean: The Second Story R` to boot and play.
+What did the trick? Allowing a {{< gh-hovercard "11952" "stereo count" >}} of zero or higher, and fixing a typo. 
+Coding shenanigans.
+
+And to finish up, what better way to end than enumerating some of the battles byte[] won for improving shutdown stability!
+
+Audio gets the first one. 
+When enough {{< gh-hovercard "12019" "stream data" >}} is still waiting in the output queue, the audio renderer will wait until it has space to add more data. 
+But during shutdown, more space might never be made available, which would block the audio renderer forever and hang. 
+Fixing this blockade fixes shutdown operations on Android.
+
+Next one is quite a bit more involved, as it involves the procedure needed to ~~kill~~ terminate a process, James Bond style.
+This procedure consists of two steps for each thread in the terminated process:
+- The thread is marked for termination, and an interrupt is sent to all cores with that thread's affinity if it is runnable.
+- The kernel then waits for the thread to actually terminate.
+
+Runnable threads marked for termination are not removed from the scheduling priority queue. This means they can get scheduled during termination, even if the core already cleared the interrupt when a previous thread was terminated.
+{{< gh-hovercard "12025" "Checking for termination" >}} just before running the thread again, as the Switch does, fixed a very common shutdown deadlock on Android.
+
+And lastly, the timing event queue was causing issues. 
+On shutdown, the system calls to clear pending events, even while the timing thread may still be removing items from the event queue.
+{{< gh-hovercard "12028" "Locking access" >}} to the event queue solves the final shutdown crash of this month.
+
+## Hardware section
+
+### AMD, mixing drivers
+
+[Last month](https://yuzu-testing.netlify.app/entry/yuzu-progress-report-oct-2023/#amd-giving-a-last-hurrah-to-polaris-and-vega) we reported how AMD is discontinuing driver support for Polaris and Vega products and how that can affect yuzu.
```

Comment:
```suggestion
[Last month](https://yuzu-emu.org/entry/yuzu-progress-report-oct-2023/#amd-giving-a-last-hurrah-to-polaris-and-vega), we reported that AMD is discontinuing driver support for Polaris and Vega products, and speculated how it would affect yuzu.
```